src/controllers/expenses.controller.js

```
@@ -0,0 +1,113 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+// Get all expenses
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  let expenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    expenses = expenses.filter((expense) => expense.userId === +userId);
```

getAll: You must validate `userId` (query) before using `+userId`. If `userId` is provided but not a valid number, return HTTP 400. Currently a non-numeric userId will silently coerce to NaN and filter nothing. Example check: `if (userId && Number.isNaN(Number(userId))) return res.status(400).send({ message: 'Invalid userId' })`. Update the code around the `userId` handling so invalid query parameters produce 400 as required by the task.